src/kernel-install/kernel-install

```diff
@@ -85,7 +85,10 @@ fi
 KERNEL_VERSION="$1"
 KERNEL_IMAGE="$2"
 
-if [[ -f /etc/machine-id ]]; then
+# Reuse directory created without a machine ID present if it exists.
+if [[ -d /efi/Linux ]] || [[ -d /boot/Linux ]] || [[ -d /boot/efi/Linux ]]; then
```

Comment:
hmm, so this is becoming a problem. Boot loader spec type `#2` entries use $ESP/EFI/Linux as directory to drop unified kernels in. This script is primarily for type `#1` entries, and the rightmost check might thus place type `#1` kernels in the type `#2` directory, if /boot happens to be the ESP... (as the ESP is FAT, i.e. case-insensitive)

This ambiguity sucks a bit. I wonder if this means we should use a a different naming here to avoid this. i.e. use "default" or so as name here?

Reply:
or even "Default" (with uppercase "D")

Reply:
Not a big fan but I can't come up with anything better either. The boot loader spec enforces that type 2 entries under $BOOT/Linux use the .efi extension but it doesn't say anything that prevents us from putting other stuff in there as well. Doesn't that give us the option to use Linux for type #1 as well?

We could go even further and put the machine id directories under the Linux folder as well so it's extremely clear for users what is Linux and what is not which is not the case at the moment. If you enter the /boot directory as a non-expert you just see a weird garbled string (the machine id) which tells you absolutely nothing about where the directory came from. This might be too big of a change though.

Reply:
> Not a big fan but I can't come up with anything better either. The boot loader spec enforces that type 2 entries under $BOOT/Linux use the .efi extension but it doesn't say anything that prevents us from putting other stuff in there as well. Doesn't that give us the option to use Linux for type #1 as well?

Well, but conversely, at least on Fedora all default kernels are built as EFI binaries, and thus if you actually use type 1 entries, they too will have the efi entry, and suddenly mght be understood as type 2 entries even though they don't work as such (since they do not include an initrd).

And besides that, it's ugly, too

Reply:
> We could go even further and put the machine id directories under the Linux folder as well so it's extremely clear for users what is Linux and what is not which is not the case at the moment. If you enter the /boot directory as a non-expert you just see a weird garbled string (the machine id) which tells you absolutely nothing about where the directory came from. This might be too big of a change though.

Yeah, for such a change we need a really good reason. I don't think "prettier" alone is sufficient and thus wouldn't bother

Reply:
Ok let's do Default.